desc/integrals/interp_utils.py

```diff
@@ -9,10 +9,395 @@
+    return fq
+
+
+def cheb_from_dct(a, axis=-1):
+    """Get discrete Chebyshev transform from discrete cosine transform.
+
+    Parameters
+    ----------
+    a : jnp.ndarray
+        Discrete cosine transform coefficients, e.g.
+        ``a=dct(f,type=2,axis=axis,norm="forward")``.
+        The discrete cosine transformation used by scipy is defined here:
+        https://docs.scipy.org/doc/scipy/reference/generated/scipy.fft.dct.html.
+    axis : int
+        Axis along which to transform.
+
+    Returns
+    -------
+    cheb : jnp.ndarray
+        Chebyshev coefficients along ``axis``.
+
+    """
+    cheb = a.copy().at[Index.get(0, axis, a.ndim)].divide(2.0)
+    return cheb
+
+
+def dct_from_cheb(cheb, axis=-1):
+    """Get discrete cosine transform from discrete Chebyshev transform.
+
+    Parameters
+    ----------
+    cheb : jnp.ndarray
+        Discrete Chebyshev transform coefficients, e.g.``cheb_from_dct(a)``.
+    axis : int
+        Axis along which to transform.
+
+    Returns
+    -------
+    a : jnp.ndarray
+        Chebyshev coefficients along ``axis``.
+
+    """
+    a = cheb.copy().at[Index.get(0, axis, cheb.ndim)].multiply(2.0)
+    return a
+
+
+def interp_dct(xq, f, lobatto=False, axis=-1):
+    """Interpolate ``f`` to ``xq`` with discrete Chebyshev transform.
+
+    Parameters
+    ----------
+    xq : jnp.ndarray
+        Real query points where interpolation is desired.
+        Shape of ``xq`` must broadcast with shape ``np.delete(f.shape,axis)``.
+    f : jnp.ndarray
+        Real function values on Chebyshev points to interpolate.
+    lobatto : bool
+        Whether ``f`` was sampled on the Gauss-Lobatto (extrema-plus-endpoint)
+        or interior roots grid for Chebyshev points.
+    axis : int
+        Axis along which to transform.
+
+    Returns
+    -------
+    fq : jnp.ndarray
+        Real function value at query points.
+
+    """
+    lobatto = bool(lobatto)
+    errorif(lobatto, NotImplementedError, "JAX hasn't implemented type 1 DCT.")
+    a = cheb_from_dct(dct(f, type=2 - lobatto, axis=axis), axis) / (
+        f.shape[axis] - lobatto
+    )
+    fq = idct_non_uniform(xq, a, f.shape[axis], axis)
+    return fq
+
+
+def idct_non_uniform(xq, a, n, axis=-1):
+    """Evaluate discrete Chebyshev transform coefficients ``a`` at ``xq`` ∈ [-1, 1].
+
+    Parameters
+    ----------
+    xq : jnp.ndarray
+        Real query points where interpolation is desired.
+        Shape of ``xq`` must broadcast with shape ``np.delete(a.shape,axis)``.
+    a : jnp.ndarray
+        Discrete Chebyshev transform coefficients.
+    n : int
+        Spectral resolution of ``a``.
+    axis : int
+        Axis along which to transform.
+
+    Returns
+    -------
+    fq : jnp.ndarray
+        Real function value at query points.
+
+    """
+    a = jnp.moveaxis(a, axis, -1)
+    # Could use Clenshaw recursion with fq = chebval(xq, a, tensor=False).
+    basis = chebvander(xq, n - 1)
```

Comment:
It's probably faster to just use `chebval` since that is basically whats done inside `chebvander` but you then skip the matvec.

The cheapest option (esp on GPU) though would be to use `T_n(x) = cos(n*arccos(x))` assuming x isn't too close to the boundary.